pkg/controller/direct/secretmanager/secret_controller.go

```diff
@@ -242,34 +242,21 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 	}
 	// the GCP service use *name* to identify the resource.
 	resource.Name = a.id.External
-
+	resource.Etag = a.actual.Etag
 	resource.Annotations = ComputeAnnotations(desired)
 	resource.Labels = common.ComputeGCPLabels(desired.GetLabels())
-
-	updateMask := &fieldmaskpb.FieldMask{}
-
-	if !reflect.DeepEqual(resource.Annotations, a.actual.Annotations) {
-		updateMask.Paths = append(updateMask.Paths, "annotations")
-	}
-	if !topicsEqual(desired.Spec.TopicRefs, a.actual.GetTopics()) {
-		updateMask.Paths = append(updateMask.Paths, "topics")
-	}
-	if !common.DeepEqual_StringAndTimestampPb(*desired.Spec.ExpireTime, a.actual.GetExpireTime()) {
-		updateMask.Paths = append(updateMask.Paths, "expire_time")
-	}
-	if !reflect.DeepEqual(desired.Spec.Rotation.NextRotationTime, a.actual.GetRotation().GetNextRotationTime()) {
-		updateMask.Paths = append(updateMask.Paths, "rotation.next_rotation_time")
-	}
-	if !reflect.DeepEqual(desired.Spec.Rotation.RotationPeriod, a.actual.GetRotation().GetRotationPeriod()) {
-		updateMask.Paths = append(updateMask.Paths, "rotation.rotation_period")
+	paths, err := common.CompareProtoMessage(resource, a.actual)
```

Comment:
We probably need the ability to provide "custom diff" function when comparing proto messages.

Reply:
Done! Thank you for the suggestion